### PR TITLE
[Program: GCI] Test PUT /cancel API for cancelling mentorship request

### LINKED ISSUE #307: Write missing tests for Cancel Mentorship Relation API
## Description
As a developer for backend api,
I need cover most part of the logic of backend,
so that I can be confident that the app works as it is supposed to.

Write tests that cover the test cases from [this table](https://github.com/systers/mentorship-backend/wiki/Quality-Assurance#cancel-relation) for the following API:
- `PUT /mentorship_relation/{request_id}/cancel`

Also write the following tests that are not in the table:
- Valid user tries to cancel a mentorship relation with authentication token missing (FAIL)
- Valid user tries to cancel a mentorship relation with authentication token expired (FAIL)

Make sure to add the test in the appropriate [tests/mentorship_relation/test_api_cancel_relation.py](https://github.com/systers/mentorship-backend/blob/develop/tests/mentorship_relation/test_api_cancel_relation.py) file.

_this issue is intended for the Google Code-in participants, it can be released after the program ends on January 23rd_

### Update [Required]
- [ ] Write missing tests for Cancel Mentorship Relation API

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by daksha-aeer**
### Description
Test cases created for **PUT /mentorship_relation/{request_id}/cancel** API:

1. User1 (sent the request) cancels a mentorship relation that it is currently involved with User2 (the relation is in an ACCEPTED state).
- Gives **200, MENTORSHIP_RELATION_WAS_CANCELLED_SUCCESSFULLY** response.
- Test case created using User1's auth header.
task33_1

2. User2 (received the request) cancels a mentorship relation that it is currently involved with User1 (the relation is in an ACCEPTED state).
- Gives **200, MENTORSHIP_RELATION_WAS_CANCELLED_SUCCESSFULLY** response.
- Test case created using User2's auth header.
task33_2

3. User 1 cancel a mentorship relation which the User1 is not involved with.
- Gives **400, CANT_CANCEL_UNINVOLVED_REQUEST** response.
- Test case created using admin_user's auth header.
task33_3

4. Valid user tries to cancel valid task with authentication token missing.
- Gives **401, AUTHORISATION_TOKEN_IS_MISSING** response.
- Test case created by not passing auth header field.
task33_4

5. Valid user tries to cancel valid task with authentication token expired.
- Gives **401, TOKEN_HAS_EXPIRED** response. 
- Test case created by generating expired auth token with help of ```token_expiration_delta```. Delta set to -10 seconds.
task33_5

Fixes #307

### Type of Change:
- Code
- Quality Assurance

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)

### How Has This Been Tested?
```sh
python -m unittest discover tests
```
task33_6

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: tests/mentorship_relation/test_api_cancel_relation.py [hunk 1]
```diff
@@ -62,6 +62,76 @@ def test__mentee_cancel_mentorship_relation(self):
...
+            response = self.client.put('/mentorship_relation/%s/cancel' % self.mentorship_relation.id,
+                                       headers=get_test_request_header(self.admin_user.id))
+            self.assertEqual(400, response.status_code)
+            self.assertEqual(MentorshipRelationState.ACCEPTED, self.mentorship_relation.state)
+            self.assertDictEqual(messages.CANT_CANCEL_UNINVOLVED_REQUEST,
+                             json.loads(response.data))
+
+    # Valid user tries to cancel valid task with authentication token missing
+    # 401, AUTHORISATION_TOKEN_IS_MISSING
+    def test__cancel_mentorship_relation_auth_token_expired(self):
```
Comment (Tlazypanda) [inline tests/mentorship_relation/test_api_cancel_relation.py]: Not the test for missing authorization token.

**Review by Tlazypanda — CHANGES_REQUESTED** [review]
@daksha-aeer I think you have accidently missed out on one of the tests :sweat_smile: Just add that and we are good to go!

File: tests/mentorship_relation/test_api_cancel_relation.py (hunk 1 above)
Comment (daksha-aeer) [reply tests/mentorship_relation/test_api_cancel_relation.py]: Fixed, please review!

**Review by Tlazypanda — APPROVED** [review]
@daksha-aeer Awesome! :fire: Approved :)

**Comment by isabelcosta** [comment]
Thank you for your contribution here @daksha-aeer 🎉 
I'm going to close this PR, since we decided to merge another one.

**Pull request closed without merging**